test/fixtures/bn.json

```diff
@@ -0,0 +1,105 @@
+{
+  "valid": [
+    {
+      "dec": "0",
+      "hex": "00"
+    },
+    {
+      "dec": "1",
+      "hex": "01"
+    },
+    {
+      "dec": "-1",
+      "hex": "ff"
```

Comment:
Agreed,  maybe it's worth having it as separate functionality?  In any
case,  I'll amend the test vectors.
On 11 Apr 2015 08:10, "Fedor Indutny" <notifications@github.com> wrote:

> In test/fixtures/bn.json
> <https://github.com/indutny/bn.js/pull/36#discussion_r28184355>:
>
> > @@ -0,0 +1,105 @@
> > +{
> > +  "valid": [
> > +    {
> > +      "dec": "0",
> > +      "hex": "00"
> > +    },
> > +    {
> > +      "dec": "1",
> > +      "hex": "01"
> > +    },
> > +    {
> > +      "dec": "-1",
> > +      "hex": "ff"
>
> @dcousens <https://github.com/dcousens> HEX is nothing special to
> .toString(), the base-36 or base-3 encoding should work the same way as
> the HEX does.
>
> I guess the two's complement thing should happen in the code responsible
> for encoding this. There are various way to encode it, and this doesn't
> look like a reason to make HEX something special.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/indutny/bn.js/pull/36/files#r28184355>.
>